src/app/clusters/scenes-server/SceneHandlerImpl.cpp

```diff
@@ -16,10 +16,224 @@
+        // wouldn't work
+        maxValue = static_cast<WorkingType>((1ULL << (bitWidth - 1)) - 1);
+        minValue = static_cast<WorkingType>(-(1ULL << (bitWidth - 1)));
+    }
+    else
+    {
+        // We use emberAfAttributeSize for cases like INT24U, INT40U, INT48U, INT56U where numeric_limits<WorkingType>::max()
+        // wouldn't work
+        if (ZCL_INT64U_ATTRIBUTE_TYPE == metadata->attributeType)
+        {
+            maxValue = static_cast<WorkingType>(UINT64_MAX); // Bit shift of 64 is undefined so we use UINT64_MAX
+        }
+        else
+        {
+            maxValue = static_cast<WorkingType>((1ULL << bitWidth) - 1);
+        }
+        minValue = static_cast<WorkingType>(0);
+    }
+
+    // Ensure that the metadata's signedness matches the working type's signedness
+    VerifyOrDie(metadata->IsSignedIntegerAttribute() == std::is_signed<WorkingType>::value);
+
+    if (metadata->IsBoolean())
+    {
+        // Caping the value to 1 in case values greater than 1 are set
+        Value = Value ? 1 : 0;
+        return;
+    }
+
+    // Check metadata for min and max values
+    if (metadata->HasMinMax())
+    {
+        const EmberAfAttributeMinMaxValue * minMaxValue = metadata->defaultValue.ptrToMinMaxValue;
+        minValue                                        = ConvertDefaultValueToWorkingValue<Type>(minMaxValue->minValue);
+        maxValue                                        = ConvertDefaultValueToWorkingValue<Type>(minMaxValue->maxValue);
+    }
+
+    if (minValue > Value)
+    {
+        Value = minValue;
+    }
+    else if (maxValue < Value)
+    {
+        Value = maxValue;
+    }
+}
+
+/// @brief  Validate the attribute exists for a given cluster
+/// @param[in] endpoint   Endpoint ID
+/// @param[in] clusterID  Cluster ID
+/// @param[in] aVPair     AttributeValuePairType, will be mutated to cap the value if it is out of range
+/// @return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute) if the attribute does not exist for a given cluster or is not scenable
+/// @note This will allways fail for global list attributes. If we do want to make them scenable someday, we will need to
+///       use a different validation method.
+// TODO: Add check for "S" quality to determine if the attribute is scenable once suported :
+// https://github.com/project-chip/connectedhomeip/issues/24177
+CHIP_ERROR ValidateAttributePath(EndpointId endpoint, ClusterId cluster, AttributeValuePairType & aVPair)
+{
+    const EmberAfAttributeMetadata * metadata = emberAfLocateAttributeMetadata(endpoint, cluster, aVPair.attributeID);
+
+    if (nullptr == metadata)
+    {
+        return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+    }
+
+    // There should never be more than one populated value in an ExtensionFieldSet
+    VerifyOrReturnError(IsOnlyOneValuePopulated(aVPair), CHIP_ERROR_INVALID_ARGUMENT);
+
+    switch (metadata->attributeType)
```

Comment:
Should this be using `AttributeBaseType(metadata->attributeType)`?  Followup fine for this, but I think it should.  And then we can have fewer cases below; right now we are including some "integer like" types in the cases but not others, which is very odd.